### PR TITLE
Feature/logic

### PR DESCRIPTION
**Pull request by alpox**
Here goes the logics

### CODE REVIEW

File: src/components/Board.tsx [hunk 1]
```diff
@@ -1,15 +1,100 @@
...
+
+interface CellProps {
+  cell: GridElement;
+}
+
+const Cell: React.FC<CellProps> = ({ cell }) => {
+  const { dispatch } = useContext(GameContext);
+  const classMapping = {
+    [GridElementState.Hidden]: "hidden",
+    [GridElementState.MarkedAsMine]: "marked",
```
Comment (bradtaniguchi) [inline src/components/Board.tsx]: Just to make sure `MarkedAsMine` means the flag right?

Comment (alpox) [reply src/components/Board.tsx]: Well in the Windows version it was a flag, yes :)

**Comment by bradtaniguchi** [comment]
Hmm when running it locally I can hit the start button and then.... nothing :open_mouth: 

image
(in the picture I hit the start button, and the number changed to 1, but that seems like it, hitting start more times doesn't do anything)

**Comment by alpox** [comment]
Looks to me like you are running the version as it was before this pull request. You havr the `master` branch state for some reason.

**Comment by bradtaniguchi** [comment]
@alpox Thats probably what happen, I tried this out late last night and probably screwed something up, I'll check it out again and double check to make sure that was the reason :+1:

**Comment by bradtaniguchi** [comment]
I just tried it out (correctly this time!) looks to be **working pretty good**. I noticed the "right click" can get a cell "stuck" in a state its darker grey and I'm not sure what that means, if the blue one is the actual "flag"?

(I have yet to win so idk how winning works haha, are there some cheat codes :wink: )

**Comment by alpox** [comment]
The other gray is `unsure`. I took that state out of some discussion in the issues. I usually dont need such a state but maybe some do otherwise it can be taken out :)

**Comment by bradtaniguchi** [comment]
I must not remembered (or was 100% of myself when playing this game :wink: ) about the unsure one, is that the `?` haha?

**Comment by alpox** [comment]
Yes it should be pretty much a questionmark. But I dont think they had that in the default windows version

**Pull request merged**